plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergHadoopConnectorSmokeTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.BaseConnectorSmokeTest;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingConnectorBehavior;
+import org.apache.iceberg.FileFormat;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.trino.plugin.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+// Redundant over TestIcebergOrcConnectorTest, but exists to exercise BaseConnectorSmokeTest
+// Some features like materialized views may be supported by Iceberg only.
+public class TestIcebergHadoopConnectorSmokeTest
+        extends BaseConnectorSmokeTest
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createIcebergQueryRunner(ImmutableMap.of(), CatalogType.HADOOP, FileFormat.ORC, REQUIRED_TPCH_TABLES, Optional.empty());
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_COMMENT_ON_COLUMN:
+            case SUPPORTS_CREATE_MATERIALIZED_VIEW:
+            case SUPPORTS_CREATE_VIEW:
+            case SUPPORTS_TOPN_PUSHDOWN:
+            case SUPPORTS_RENAME_TABLE:
+                return false;
+            case SUPPORTS_RENAME_TABLE_ACROSS_SCHEMAS:
+            case SUPPORTS_DELETE:
+                return true;
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Test
+    @Override
+    public void testRowLevelDelete()
+    {
+        // Deletes are covered AbstractTestIcebergConnectorTest
+        assertThatThrownBy(super::testRowLevelDelete)
+                .hasStackTraceContaining("This connector only supports delete where one or more partitions are deleted entirely");
+    }
+
+    @Test
+    @Override
+    public void testRenameTable()
+    {
+        assertThatThrownBy(super::testRenameTable)
+                .hasStackTraceContaining("Cannot rename Hadoop tables");
```

Comment:
this error message is not descriptive enough.

Reply:
This is the error message defined in Iceberg: https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java#L277